llvm/test/Transforms/PGOProfile/memprof_direct_recursion.ll

```diff
@@ -0,0 +1,132 @@
+;; Tests memprof when contains recursion.
```

Comment:
The tests in this directory are typically just checking matching, but here we are testing the generated ThinLTO summary. It would probably be better to have this test case be in llvm/test/ThinLTO/X86. No need to test the matching, just use the IR with the matched summary metadata as input and test the summary generation - you can also easily force the context to be cold that way. And if we can possibly just fix the assertion and not remove the direct recursion handling, as suggested in my previous comment, also check the cloning we get.